src/clue.js

```diff
@@ -4,23 +4,222 @@
+}
+
+const livingRoom = {
+    name: "Living Room"
+}
+const observatory = {
+    name: "Observatory"
+}
+const theater = {
+    name: "Theater"
+}
+
+const guestHouse = {
+    name: "Guest House"
+}
+const patio = {
+    name: "Patio"
+}
+
+roomsArray.push(diningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio)
+//console.log(roomsArray)
+
 // Weapons Array
 
 const weaponsArray = [];
 
+const rope = {
+    name: "rope",
+    weight: 10
+}
+const knife = {
+    name: "knife",
+    weight: 8
+}
+
+const candlestick={
+    name: "candlestick",
+    weight: 2
+}
+
+const dumbbell={
+    name: "dumbbell",
+    weight: 30
+}
+
+const poison ={
+    name: "poison",
+    weight: 2
+}
+
+const axe = {
+    name: "axe",
+    weight: 15
+}
+const bat={
+    name: "bat",
+    weight: 13
+}
+const trophy={
+    name: "trophy",
+    weight: 25 
+}
+const pistol={
+    name: "pistol",
+    weight: 20
+}
+
+weaponsArray.push(rope, knife, candlestick,dumbbell, poison, axe, bat, trophy, pistol)
+//console.log(weaponsArray)
 
 // ITERATION 2
+// Declare a function named selectRandom to select one element from a card stack randomly. The function should expect an array as an argument and should return a random element from the array.
+
+function selectRandom(array) {
+    if(array.length === 0){
+        return undefined
+    }
+    return array[Math.floor(Math.random()*array.length)]
+}
+
+// Declare a function named pickMystery that takes no arguments and returns an object with three properties: suspect, weapon and room, each holding as a value a card of that specific type. You can get a random card of each type by calling selectRandom on each card stack.
 
-function selectRandom() {}
+function pickMystery() {
+    let returnedObject = {}
+    returnedObject.suspect = selectRandom(suspectsArray);
+    returnedObject.weapon = selectRandom(weaponsArray);
+    returnedObject.room = selectRandom(roomsArray);
 
-function pickMystery() {}
+    return returnedObject;
+}
 
+console.log(pickMystery());
 
 // ITERATION 3
+// Declare a function named revealMystery that receives an envelope object (with the shape of the object returned by pickMystery) as the single argument and returns a revealing message in the following format:
+
+// <FIRST NAME> <LAST NAME> killed Mr. Boddy using the <WEAPON> in the <ROOM>!
+
+const obj1 = {
+    suspect: {
+      firstName: 'Doctor',
+      lastName: 'Orchid',
+      occupation: 'Scientist',
+      age: 26,
+      description: 'PhD in plant toxicology. Adopted daughter of Mr. Boddy',
+      image: 'http://www.radiotimes.com/uploads/images/Original/111967.jpg',
+      color: 'white'
+    },
+    weapon: { name: 'dumbbell', weight: 30 },
+    room: { name: 'Spa' }
+  }
+
+function revealMystery(obj) {
+    return `${obj.suspect.firstName} ${obj.suspect.lastName} killed Mr. Boddy using the ${obj.weapon.name} in the ${obj.room.name}!`
+}
```

Comment:
Here, you have everything correct, however, it is good practice if you destructure your `obj` argument.
By destructuring, it clearly expresses which properties you're using within the function.
By extracting the properties into individual variables (suspect, weapon, and room), you can directly reference them in your return statement without needing to use `obj.suspect`, `obj.weapon`, and `obj.room` every time. This simplifies the code and improves readability.
It would look something like this:

```
function revealMystery(obj) {
  let { suspect, weapon, room } = obj;

  return `${suspect.firstName} ${suspect.lastName} killed Mr. Boddy using the ${weapon.name} in the ${room.name}!`;
}
```